pymc/distributions/distribution.py

```diff
@@ -164,6 +165,45 @@ def fn(*args, **kwargs):
     return fn
 
 
+# Helper function from pyprob
+def _extract_target_of_assignment(depth):
+    frame = sys._getframe(depth)
+    code = frame.f_code
+    next_instruction = code.co_code[frame.f_lasti + 2]
+    instruction_arg = code.co_code[frame.f_lasti + 3]
+    instruction_name = opcode.opname[next_instruction]
+    if instruction_name == "STORE_FAST":
+        return code.co_varnames[instruction_arg]
+    elif instruction_name in ["STORE_NAME", "STORE_GLOBAL"]:
+        return code.co_names[instruction_arg]
+    elif (
+        instruction_name in ["LOAD_FAST", "LOAD_NAME", "LOAD_GLOBAL"]
+        and opcode.opname[code.co_code[frame.f_lasti + 4]] in ["LOAD_CONST", "LOAD_FAST"]
+        and opcode.opname[code.co_code[frame.f_lasti + 6]] == "STORE_SUBSCR"
+    ):
+        base_name = (code.co_varnames if instruction_name == "LOAD_FAST" else code.co_names)[
+            instruction_arg
+        ]
+        second_instruction = opcode.opname[code.co_code[frame.f_lasti + 4]]
+        second_arg = code.co_code[frame.f_lasti + 5]
+        if second_instruction == "LOAD_CONST":
+            value = code.co_consts[second_arg]
+        elif second_instruction == "LOAD_FAST":
+            var_name = code.co_varnames[second_arg]
+            value = frame.f_locals[var_name]
+        else:
+            value = None
+        if type(value) is int:
```

Comment:
Is this for `container[i] = pm.Normal(0, 1)`?

Missing a test if that's the case.

Reply:
I haven't tested assignment in a list comprehension. Can you include a short example?

Lots of the complexity in `_extract_target_of_assignment` is for supporting a wider variety of things you can assign to like `container[i]`.

Reply:
Here I was asking what this integer case was.

For list comprehension I was thinking `normals = [pm.Normal() for i in range(5)]`

Not saying we need to support, just asking if it would be possible.

Reply:
The integer case is for assigning to tuples I believe. The list comprehension case does not work, but we could probably modify `_extract_target_of_assignment` to detect that situation.

Reply:
I think we should hold off on supporting it. Generating unique names for comprehensions that are predictable to the user looks tricky to me.

Reply:
> The integer case is for assigning to tuples I believe.

Can you add a test?